docs/guides/tfhers/shared-key.md

```diff
@@ -0,0 +1,247 @@
+# Scenario 1: Shared Secret Key
+
+The first scenario involves a single party seeking to combine both Concrete and TFHE-rs in a computation. In this case, a shared secret key will be used, while different keysets will be held for Concrete and TFHE-rs. It is possible to generate keys in both directions. In case 1.1, the steps are as follows:
+
+1. We do a classical keygen in Concrete, which generates a bunch of secret keys (we would be interested in the one used for TFHE-rs) and public keys
+2. We will use the secret key meant for TFHE-rs to launch a partial keygen (i.e. starting from a secret key and generating the rest of the keys) in TFHE-rs
+
+Case 1.2 is basically the opposite, the steps are as follows:
+
+1. We do a classical keygen in TFHE-rs, which generates a single secret key and different public keys
+2. We will use the secret key to launch a partial keygen in Concrete
+
+After the key generation is complete and we have both keysets, we can perform computations, encryption, and decryption on both ends.
+
```

Comment:
Suggestions:

## Key generation 
In this scenario, a shared secret key will be used, while different keysets will be held for Concrete and TFHE-rs. There are two ways to generate keys, outlined with the following steps
### From Concrete
1. Perform a classical key generation in Concrete, which generates a set of secret and public keys.
2. Identify the secret key intended for TFHE-rs.
3. Use this secret key to perform a partial key generation in TFHE-rs, starting from the shared secret key and generating the rest of the necessary keys.

### From TFHE-rs
1. Perform a classical key generation in TFHE-rs, generating a single secret key and corresponding public keys.
2. Use the secret key from TFHE-rs to perform a partial keygen in Concrete.

After the key generation is complete and we have both keysets, we can perform computations, encryption, and decryption on both ends.